ql/src/java/org/apache/hadoop/hive/ql/io/orc/VectorizedOrcAcidRowBatchReader.java

```diff
@@ -232,6 +250,17 @@ private VectorizedOrcAcidRowBatchReader(JobConf conf, OrcSplit orcSplit, Reporte
 
     this.syntheticProps = orcSplit.getSyntheticAcidProps();
 
+    if (LlapHiveUtils.isLlapMode(conf) && LlapProxy.isDaemon()
+            && HiveConf.getBoolVar(conf, ConfVars.LLAP_TRACK_CACHE_USAGE))
+    {
+      MapWork mapWork = LlapHiveUtils.findMapWork(conf);
```

Comment:
We could spare the deserialization of MapWork from JobConf here, if we pass the MapWork instance already present in LlapRecordReader to VectorizedOrcAcidRowBatchReader ctor. (Downside is that in turn we would need to adjust the other ctor's of VectorizedOrcAcidRowBatchReader too)

Reply:
Good idea, done!